libyul/optimiser/StackLimitEvader.cpp

```diff
@@ -66,12 +67,37 @@ struct MemoryOffsetAllocator
 		if (unreachableVariables.count(_function))
 		{
 			yulAssert(!slotAllocations.count(_function), "");
+
+			if (functionDefinitions.count(_function))
+			{
+				FunctionDefinition const* functionDefinition = functionDefinitions.at(_function);
+				yulAssert(functionDefinition, "");
+				size_t totalArgCount = functionDefinition->returnVariables.size() + functionDefinition->parameters.size();
+				size_t slotsNeeded = (totalArgCount > 16) ? totalArgCount - 16 : 0;
```

Comment:
I wonder if we should have a `const Instruction::stackAccessDepth = 16;` in `libevmasm/Instruction.h`, since we use this number so frequently in a ton of places. Having a constant may help in finding all stack related code places.